azurerm/internal/services/containers/container_registry_scope_map_resource.go

```diff
@@ -0,0 +1,230 @@
+	}
+
+	read, err := client.Get(ctx, resourceGroup, containerRegistryName, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving scope map %q for Container Registry %q (Resource Group %q): %+v", name, containerRegistryName, resourceGroup, err)
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read scope map %q for Container Registry %q (resource group %q) ID", name, containerRegistryName, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceContainerRegistryScopeMapRead(d, meta)
+}
+
+func resourceContainerRegistryScopeMapUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Containers.ScopeMapsClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for AzureRM Container Registry scope map update.")
+	resourceGroup := d.Get("resource_group_name").(string)
+	containerRegistryName := d.Get("container_registry_name").(string)
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	actions := d.Get("actions").([]interface{})
+
+	parameters := containerregistry.ScopeMapUpdateParameters{
+		ScopeMapPropertiesUpdateParameters: &containerregistry.ScopeMapPropertiesUpdateParameters{
+			Description: utils.String(description),
+			Actions:     utils.ExpandStringSlice(actions),
+		},
+	}
+
+	future, err := client.Update(ctx, resourceGroup, containerRegistryName, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error updating scope map %q for Container Registry %q (Resource Group %q): %+v", name, containerRegistryName, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for update of scope map %q (Container Registry %q, Resource Group %q): %+v", name, containerRegistryName, resourceGroup, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, containerRegistryName, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving scope map %q (Container Registry %q, Resource Group %q): %+v", name, containerRegistryName, resourceGroup, err)
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read scope map %q (Container Registry %q, resource group %q) ID", name, containerRegistryName, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceContainerRegistryScopeMapRead(d, meta)
+}
+
+func resourceContainerRegistryScopeMapRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Containers.ScopeMapsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ContainerRegistryScopeMapID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	containerRegistryName := id.RegistryName
+	name := id.ScopeMapName
+
+	resp, err := client.Get(ctx, resourceGroup, containerRegistryName, name)
```

Comment:
could we use the id var instead?
```suggestion

	resp, err := client.Get(ctx, id.ResourceGroup, id.ContainerRegistryName, id.Name)
```